grammar.js

```diff
@@ -274,8 +274,7 @@ module.exports = grammar({
       seq(
         $._indent,
         optional(seq($.shebang, $._newline)),
-        repeat(choice(seq($.recipe_line, $._newline), $._newline)),
-        $._newline,
+        repeat1(seq($.recipe_line, $._newline)),
```

Comment:
Hey, what happens if instead of changing to `repeat1` you just drop the newline on line 278 that's part of the parent `seq`?

These two patterns aren't equivalent if newlines get tokenized separately - the original one enforces (`line\n` but it can be interspersed with just `\n`, whereas the one in this PR doesn't allow plain newlines within the rule.

It shouldn't make a difference if `\r\n` is always treated idential to `\n` (i.e. doesn't count as a token), but maybe there is a bug in the scanner where that isn't the case?

Reply:
I would have though that the grammar for `recipe_body` permitting a _zero_ or more newline terminal node/token would be an issue, but for some reason it isn't 🤷? 
<img width="221" alt="image" src="https://github.com/IndianBoy42/tree-sitter-just/assets/11223234/a645cf08-883b-403a-bea6-00be10565214">

Change:
https://github.com/IndianBoy42/tree-sitter-just/blob/9dfcd6a10e02b306c8c41fff09d07c02b159c018/grammar.js#L273-L279

Reply:
Still fails, hmm. Maybe this?

```js
        $._indent,
        optional(seq($.shebang, $._newline)),
        repeat(choice(seq($.recipe_line, $._newline), $._newline)),
        $._dedent,
```

Reply:
I'll try, could part of the reason be that we're not handling optional carriage returns in the newline lookahead?
https://github.com/IndianBoy42/tree-sitter-just/blob/5d71db22c086fdb21fe12adba4b9167073e0f389/src/scanner.c#L162-L170

Reply:
Awesome!!

Could you rebase & squash? GH is refusing to do it, says there are conflicts, maybe you rebased onto an outdated main by accident. (If you don't get to it I can do this later)

Reply:
Just squashed, apologies for the delay. LMK if there are still issues.